### PR TITLE
MSPSDS-566 Use redis for session data storage

### PR DESCRIPTION
**Pull request by jasiekmiko**
The connection of this change to the msa flow lies in the amount of data
the msa flow wants to store in the session before committing everything
to the db at the end of it. We started running out of space in the 4kb
of the cookie.

Additionally, server-side session stores have
some security advantages over client-side:
https://www.happybearsoftware.com/almost-protect-yourself-from-cookie-session-store

### CODE REVIEW

**Comment by jasiekmiko** [comment]
TODO:
[] Set up new redis instance on int
[] Set up new redis instance on staging
[] Set up new redis instance on prod

File: mspsds-web/config/redis_queue.yml [hunk 1]
```diff
@@ -8,4 +9,4 @@ test:
   <<: *default
 
 production:
-  url: <%= ENV["VCAP_SERVICES"] && CF::App::Credentials.find_by_service_name("mspsds-redis")["uri"] %>
+  url: <%= ENV["VCAP_SERVICES"] && CF::App::Credentials.find_by_service_name("mspsds-redis-queue")["uri"] %>
```
Comment (jasiekmiko) [inline mspsds-web/config/redis_queue.yml]: @broder I imagine for this I'll need to create new service instances and, once deployed, delete the old ones. Is there something to be done to not loose pending jobs here?

Comment (broder) [reply mspsds-web/config/redis_queue.yml]: Would https://cli.cloudfoundry.org/en-US/cf/rename-service.html be useful here?

Comment (jasiekmiko) [reply mspsds-web/config/redis_queue.yml]: 🤦‍♂️ yes! :P
I thought we can't rename a bound service and wasn't sure if unbinding while it's running would be OK to do, but I must have confused the limitations with that of upgrading. Thanks!

File: mspsds-web/README.md [hunk 2]
```diff
@@ -130,9 +130,11 @@ To create an Elasticsearch instance for the current space:
 To create a redis instance for the current space. 
 
     cf marketplace -s redis
-    cf create-service redis tiny-unclustered-3.2 mspsds-redis
+    cf create-service redis tiny-unclustered-3.2 mspsds-redis-queue
```
Comment (broder) [inline mspsds-web/README.md]: I don't think `tiny-unclustered-3.2` exists anymore:

```
Hi,

We have released a new Redis plan - medium-ha-3.2. This provides 6.38GiB memory and provides a hot standby service for failover in the event that the original service fails.

We have also enabled backups for our tiny-3.2 plan for all instances created from now on. Please contact support at gov-uk-paas-support@digital.cabinet-office.gov.uk if you would like backups enabled for your existing instance(s).

In light of these changes we're deprecating our tiny-unclustered-3.2 plan. All new instances should be created using the tiny-3.2 plan instead.

Please contact gov-uk-paas-support@digital.cabinet-office.gov.uk if you have any questions.

Regards,
GOV.UK PaaS Team
```

Hopefully that doesn't have much impact for us.

File: mspsds-web/manifest.yml [hunk 3]
```diff
@@ -10,5 +10,6 @@ applications:
   services:
     - mspsds-database
     - mspsds-elasticsearch
-    - mspsds-redis
+    - mspsds-redis-queue
```
Comment (broder) [inline mspsds-web/manifest.yml]: I wonder if calling them `mspsds-queue` and `mspsds-session` would be better. Something about decoupling the function from the implementation?

Comment (jasiekmiko) [reply mspsds-web/manifest.yml]: Yeah, sounds good.

File: mspsds-web/README.md (hunk 2 above)
Comment (jasiekmiko) [reply mspsds-web/README.md]: hmm, it's not obvious from this if `tiny-3.2` is clustered? if it is, then the comment below that sidekiq doesn't work with clustered is a bit worrying...

Comment (jasiekmiko) [reply mspsds-web/README.md]: cf lists what we've got currently as `tiny-3.2` so it looks like those got moved over automatically, / it's just a name convention change ..

**Pull request merged**